src/app/context/theme.context.tsx

```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import React, { Dispatch, ReactNode, SetStateAction, createContext, useContext, useMemo, useState } from 'react';
+import cn from 'classnames';
+
+interface TTheme {
+  theme: 'light' | 'dark';
+}
+
+interface TThemeContext extends TTheme {
+  setTheme: Dispatch<SetStateAction<'light' | 'dark'>>;
+}
+
+const ThemeContext = createContext<TThemeContext>({ theme: 'light', setTheme: () => {} });
+
+export const useThemeContext = () => useContext(ThemeContext);
+
+function ThemeProvider({ children }: { children: ReactNode }) {
+  const [theme, setTheme] = useState<'light' | 'dark'>('light');
+
+  const themeContextValue = useMemo(() => ({ theme, setTheme }), [theme, setTheme]);
+
+  return (
+    <ThemeContext.Provider value={themeContextValue}>
+      <body className={cn({ dark: theme === 'dark' })}>{children}</body>
```

Comment:
Please remove the `body` from the context. If you need to toggle the class on the `body` it’d be better to do in the layout or any other places

Reply:
I think I can't toggle class on the `body` in `layout.tsx`, because I need `useThemeContext` there, but it's server component. So I put `body` in `App.tsx`.